app/uk/gov/hmrc/soletraderidentificationfrontend/controllers/NinoIVController.scala

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.soletraderidentificationfrontend.controllers
+
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import uk.gov.hmrc.auth.core.retrieve.v2.Retrievals.internalId
+import uk.gov.hmrc.auth.core.{AuthConnector, AuthorisedFunctions}
+import uk.gov.hmrc.http.InternalServerException
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendController
+import uk.gov.hmrc.soletraderidentificationfrontend.services._
+
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.ExecutionContext
+
+@Singleton
+class NinoIVController @Inject()(mcc: MessagesControllerComponents,
+                                 val authConnector: AuthConnector,
+                                 ninoIVService: NinoIVService,
+                                 journeyService: JourneyService,
+                                 registrationOrchestrationService: RegistrationOrchestrationService,
+                                 soleTraderIdentificationService: SoleTraderIdentificationService
+                                )(implicit val executionContext: ExecutionContext) extends FrontendController(mcc) with AuthorisedFunctions {
+
+  def retrieveNinoIVResult(journeyId: String): Action[AnyContent] = Action.async {
+    implicit req =>
+      authorised().retrieve(internalId) {
+        case Some(authInternalId) =>
+          req.getQueryString("journeyId") match {
+            case Some(verificationJourneyId) =>
+              for {
+                journeyConfig <- journeyService.getJourneyConfig(journeyId, authInternalId)
+                verificationStatus <- ninoIVService.retrieveNinoIVStatus(verificationJourneyId)
+                _ <- soleTraderIdentificationService.storeBusinessVerificationStatus(journeyId, verificationStatus)
```

Comment:
is it possible that this status will be overwritten, for example will be Fail from Nino IV, then we go back and do not provide Nino and through BV journey and get Pass? are there any implications?

Reply:
Yeah, I guess so. I'm not sure we can do much to prevent that. I guess if they were stored/returned as two separate statuses it wouldn't override it but I think it's come from the business side. Potentially one to look out for and we could have an argument to update it if it is an issue